src/libraries/System.IO.Compression/src/System/IO/Compression/ZipArchiveEntry.cs

```diff
@@ -191,14 +216,11 @@ private set
                 if (value == null)
                     throw new ArgumentNullException(nameof(FullName));
 
-                bool isUTF8;
-                _storedEntryNameBytes = EncodeEntryName(value, out isUTF8);
-                _storedEntryName = value;
+                _storedEntryNameBytes = ZipHelper.GetEncodedTruncatedBytesFromString(
+                    value, _archive.EntryNameAndCommentEncoding, 0 /* No truncation */, out bool hasUnicodeEntryName);
 
-                if (isUTF8)
-                    _generalPurposeBitFlag |= BitFlagValues.UnicodeFileName;
-                else
-                    _generalPurposeBitFlag &= ~BitFlagValues.UnicodeFileName;
```

Comment:
Yes. This bit flag's value gets changed by both the `FullName` and the `Comment` now. Instead of keeping this logic here, and adding similar logic to the `Comment` setter, I am setting the boolean value with an `|=`, and then checking the value of the boolean in `WriteCentralDirectoryFileHeader` before writing.